homeassistant/components/enigma2/media_player.py

```diff
@@ -66,8 +66,7 @@ def __init__(self, coordinator: Enigma2UpdateCoordinator) -> None:
         super().__init__(coordinator)
 
         self._attr_unique_id = (
-            coordinator.device.mac_address
-            or cast(ConfigEntry, coordinator.config_entry).entry_id
+            coordinator.device.mac_address or coordinator.config_entry.entry_id
```

Comment:
This change should be in a separate PR IMO
```suggestion
            coordinator.device.mac_address
            or cast(ConfigEntry, coordinator.config_entry).entry_id
```